lxd/main_netcat.go

```diff
@@ -21,27 +24,83 @@ func cmdNetcat(args []string) error {
 		return fmt.Errorf("Bad arguments %q", args)
 	}
 
+	// The socket name on LXD instances prior to netcat log file is:
+	// 	/<some-tmp-path/lxd_rsync_<random-suffix>
+	// and on LXD instances with netcat log file:
+	// 	/<some-tmp-path/lxd_<ctName>_rsync_<random-suffix>
+	// Detect the latter case and create a netcat log
+	var logFile *os.File
+	var err error
+	ctName := ""
+	logPath := ""
+	namestart := strings.LastIndex(args[1], "/lxd_")
```

Comment:
My comment wasn't so much about performance as about redability and simplicity. There's quite a lot of code here to extract a substring from that string.

Maybe even just extracting that logic to a separate function would make the code clearer